dts/arm/nxp/nxp_imx.dtsi

```diff
@@ -0,0 +1,223 @@
+			device_type = "cpu";
+			compatible = "arm,cortex-m4";
+			reg = <0>;
+		};
+	};
+
+	soc {
+
+		ddr_code: code@10000000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x10000000 0xfff0000>;
+			label = "DDR CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ddr_sys: memory@80000000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x80000000 0x60000000>;
+			label = "DDR SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		tcml_code: code@1fff8000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x1fff8000 0x8000>;
+			label = "TCML CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		tcmu_sys: memory@20000000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x20000000 0x8000>;
+			label = "TCMU SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_code: code@00900000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x00900000 0x20000>;
+			label = "OCRAM CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_sys: memory@20200000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x20200000 0x20000>;
+			label = "OCRAM SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_s_code: code@20180000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x20180000 0x8000>;
+			label = "OCRAM_S CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_s_sys: memory@00000000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x00000000 0x8000>;
+			label = "OCRAM_S SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_pxp_code: code@00940000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x00940000 0x8000>;
+			label = "OCRAM_PXP CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_pxp_sys: memory@20240000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x20240000 0x8000>;
+			label = "OCRAM_PXP SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_epdc_code: code@00920000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x00920000 0x20000>;
+			label = "OCRAM_EPDC CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_epdc_sys: memory@20220000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x20220000 0x20000>;
+			label = "OCRAM_EPDC SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		gpio1: gpio@30200000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x30200000 0x10000>;
+			interrupts = <62 0>, <63 0>;
+			label = "GPIO_1";
+		};
+
+		/* For now only uart2 is supported and
+		 * tested with the serial driver
+		 */
+		uart1: uart@30860000 {
+			compatible = "nxp,imx-uart";
+			reg = <0x30860000 0x10000>;
+			interrupts = <26 0>;
+			interrupt-names = "status";
+			zephyr,irq-prio = <3>;
+			label = "UART_1";
+			status = "disabled";
+		};
+
+		uart2: uart@30890000 {
```

Comment:
The address should be 30870000

Reply:
I think that the right address value is 0x30890000. Looking at the i.MX7D reference manual on page 198 it mentions 0x30870000 but on page 4534 it mentions 0x30890000. I tested with 0x30870000 and couldn't get the UART working.

Reply:
The cmsis header also uses 0x30890000, so it looks like you're right and there's an error in the reference manual. I'll let our docs team know.